Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs

```diff
@@ -0,0 +1,219 @@
+using System.Linq;
+using Content.Shared._DV.Weapons.Ranged.Components;
+using Content.Shared.Actions;
+using Content.Shared.DoAfter;
+using Content.Shared.Hands;
+using Content.Shared.Hands.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Maps;
+using Content.Shared.Toggleable;
+using Content.Shared.Weapons.Ranged.Events;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Weapons.Ranged.Systems;
+
+public abstract partial class SharedGunSystem
+{
+    [Dependency] private readonly ActionContainerSystem _actionContainer = default!;
+
+    private void InitializeBipods()
+    {
+        SubscribeLocalEvent<GunBipodComponent, GotUnequippedHandEvent>(OnUnequip);
+        SubscribeLocalEvent<GunBipodComponent, DroppedEvent>(OnDrop);
+
+        SubscribeLocalEvent<GunBipodComponent, GetItemActionsEvent>(OnGetActions);
+        SubscribeLocalEvent<GunBipodComponent, ToggleActionEvent>(OnToggleAction);
+
+        SubscribeLocalEvent<GunBipodComponent, MapInitEvent>(OnMapInit);
+
+        SubscribeLocalEvent<GunBipodComponent, GunRefreshModifiersEvent>(OnGunRefreshModifiers);
+        SubscribeLocalEvent<GunBipodComponent, BipodSetupFinishedEvent>(SetupBipod);
+        SubscribeLocalEvent<GunBipodComponent, ShotAttemptedEvent>(OnShotAttempted);
+    }
+
+    private void OnMapInit(EntityUid uid, GunBipodComponent component, MapInitEvent args)
+    {
+        _actionContainer.EnsureAction(uid, ref component.BipodToggleActionEntity, component.BipodToggleAction);
+        Dirty(uid, component);
+    }
+
+    private void OnUnequip(Entity<GunBipodComponent> ent, ref GotUnequippedHandEvent args)
+    {
+        PackUpBipod(ent, args.User);
+    }
+
+    private void OnDrop(Entity<GunBipodComponent> ent, ref DroppedEvent args)
+    {
+        PackUpBipod(ent, args.User);
+    }
+
+    private void OnGetActions(Entity<GunBipodComponent> ent, ref GetItemActionsEvent args)
+    {
+        args.AddAction(ref ent.Comp.BipodToggleActionEntity, ent.Comp.BipodToggleAction);
+    }
+
+    private void OnToggleAction(Entity<GunBipodComponent> ent, ref ToggleActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var blockQuery = GetEntityQuery<GunBipodComponent>();
+        var handQuery = GetEntityQuery<HandsComponent>();
+
+        if (!handQuery.TryGetComponent(args.Performer, out var hands))
+            return;
```

Comment:
No need to get blockQuery if the handQuery fails
```suggestion
        var handQuery = GetEntityQuery<HandsComponent>();

        if (!handQuery.TryGetComponent(args.Performer, out var hands))
            return;

        var blockQuery = GetEntityQuery<GunBipodComponent>();
```

Reply:
Rewrote the code to allow multiple Bipods to be used.